app/services/solidus_pay_tomorrow/handle_order_creation_service.rb

```diff
@@ -25,8 +25,10 @@ def create_order_on_pay_tomorrow
     def create_payment
       return unless order_successfully_created?
 
-      order.payments.create!(payment_method: payment_method, state: :checkout, response_code: @result['token'],
-        amount: @order.total)
+      source = SolidusPayTomorrow::PaymentSource.create!(application_token: @result['token'],
+        payment_method: payment_method)
+      order.payments.create!(payment_method: payment_method, source: source, state: :checkout,
+        response_code: @result['token'], amount: @order.total)
```

Comment:
If the payment creation fails, the PaymentSource is not rolled-back. Can I suggest you to add both creation into one transaction or try to use them together with:

```
source = SolidusPayTomorrow::PaymentSource.new(application_token: @result['token'],
  payment_method: payment_method)
order.payments.create!(payment_method: payment_method, source: source, state: :checkout,
  response_code: @result['token'], amount: @order.total)
```

Reply:
Fixed in https://github.com/nebulab/solidus_pay_tomorrow/pull/12/commits/b730d737b7e40c5de2ada26c95ffe8afa3a900a5!